tummensabot/mensa.py

```diff
@@ -0,0 +1,242 @@
+        enabled = args[1] == "1"
+        change_notifications(message.chat, presel, enabled)
+
+        if enabled:
+            noti_btn = InlineKeyboardButton("Auto-Update deaktivieren", callback_data="5$0")
+            msg = "Auto-Update aktiviert für Mensa " + MENSEN[int(presel)]
+        else:
+            noti_btn = InlineKeyboardButton("Auto-Update aktivieren", callback_data="5$1")
+            msg = "Auto-Update deaktiviert"
+
+        reply_markup = InlineKeyboardMarkup([[noti_btn]] + button_list)
+        send(context.bot, message.chat_id, msg, reply_markup=reply_markup, message_id=message.message_id)
+    else:
+        logging.error("unknown inline command")
+        msg = "Kommando nicht erkannt"
+        send(context.bot, message.chat_id, msg)
+        dev_msg = f"Inlinekommando nicht erkannt.\n\nData: {update.callback_query.data}\nUser:{message.chat}"
+        send_developer_message(context.bot, dev_msg)
+
+
+def send_notifications(bot=None):
+    if bot is None:
+        bot = Bot(token=config['BotToken'])
+
+    plans = {}
+    for mensa_id, mensa_name in MENSEN.items():
+        print(f"Getting plan for {mensa_name} (#{mensa_id})")
+        plans[mensa_id] = menu_manager.get_menu(mensa_id)
+
+    noti_btn = InlineKeyboardButton("Auto-Update deaktivieren", callback_data="5$0")
+    reply_markup = InlineKeyboardMarkup([[noti_btn]] + button_list)
+
+    session = Session()
+    users = session.query(User).filter(User.notifications > 0)
+    for user in users:
+        user.counter += 1
+        session.commit()
+        try:
+            print("Sending plan to", user.first_name)
+            send(bot, user.id, plans[int(user.notifications)].get_meals_message(),
+                 reply_markup=reply_markup, message_id=user.message_id)
+        except TypeError:
+            logging.exception(f"Caught TypeError while processing user {user.first_name}")
+
+    session.close()
+
+
+def job_callback(context: CallbackContext):
+    logging.debug("Scheduled notification update triggered")
+    send_notifications(bot=context.bot)
+
+
+def run_daemon():
+    updater = Updater(token=config["BotToken"], use_context=True)
+    dispatcher = updater.dispatcher
+
+    start_handler = CommandHandler('start', start)
+    dispatcher.add_handler(start_handler)
+    about_handler = CommandHandler('about', about)
+    dispatcher.add_handler(about_handler)
+    inline_handler = CallbackQueryHandler(inline_callback)
+    dispatcher.add_handler(inline_handler)
+    fallback_handler = MessageHandler(Filters.all, start)
+    dispatcher.add_handler(fallback_handler)
+
+    # schedule daily update
+    hour = int(config.get('NotificationHour', 16))
+    first = datetime.time(hour=hour, minute=0)
+    now = datetime.datetime.now()
+    logging.info(f"Job will run daily at {first}. Server time is {now.strftime('%H:%M:%S')}.")
+    updater.job_queue.run_daily(job_callback, time=first)
+
+    webhook_url = config.get('WebhookUrl', "").strip()
```

Comment:
can you just for convenience call `updater.bot.set_webhook(webhook_url)` to register the endpoint at the Telegram servers?